source/sdk/io/native/FileUnix.ooc

```diff
@@ -181,18 +188,22 @@ version (unix || apple) {
 		 * current user, group, and other.
 		 */
 		setExecutable: override func (exec: Bool) -> Bool {
-			result: FileStat
-			res := lstat(path as CString, result&)
+			lstatResult, fstatResult: FileStat
+			res := lstat(path as CString, lstatResult&)
 			if (res != 0) return false // couldn't get file mode
-
-			mode := result st_mode
+			fd := open(path as CString, READWRITE)
+			if (fd == -1) return false
+			res = fstat(fd, fstatResult&)
+			if (res != 0) return false
+			if (fstatResult st_ino != lstatResult st_ino || fstatResult st_mode != lstatResult st_mode)
```

Comment:
Fix is to use `lstat(path), fd := open(path), fstat(fd)` and compare results of both `*stat` calls. If the file was changed between calls to `lstat` and `fstat`, then it is not safe to set the executable permission.